scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/edf/EDFPolicy.java

```diff
@@ -82,6 +84,12 @@
      * (this is how this policy treats absence of deadline)
      */
     private static Date getDeadLineFromJob(InternalJob internalJob) {
-        return internalJob.getJobDeadline().orElse(new Date(Long.MAX_VALUE));
+        if (internalJob.getJobDeadline().isPresent()) {
+            if (internalJob.getJobDeadline().get().isAbsolute()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)